### PR TITLE
Update "tomcrypt" to v1.18.2

### PR DESCRIPTION
**Pull request by t0p4**
* I've followed [this guide](https://docs.hunter.sh/en/latest/creating-new/update.html)
  step by step carefully. **[Yes]**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":

  * https://ci.appveyor.com/project/t0p4/hunter/builds/19324550
  * https://travis-ci.com/t0p4/hunter/builds/87132697

---

### CODE REVIEW

File: cmake/projects/tomcrypt/hunter.cmake [hunk 1]
```diff
@@ -41,6 +41,16 @@ hunter_add_version(
     SHA1
     69b882eff5bf69a5a4e0e9169cc9b9c0ea5b9d71
 )
+hunter_add_version(
+    PACKAGE_NAME
+    tomcrypt
+    VERSION
+    "1.18.2"
+    URL
+    "https://github.com/t0p4/libtomcrypt/archive/v1.18.2-p.tar.gz"
```
[the remark this answers is not in the document]
Comment (t0p4) [reply cmake/projects/tomcrypt/hunter.cmake]: Could you give me more details on how to do it correctly?

[the remark this answers is not in the document]
Comment (ruslo) [reply cmake/projects/tomcrypt/hunter.cmake]: Start with `hunter-1.18.2` branch and cherry-pick patches from previous version:
* https://github.com/hunter-packages/libtomcrypt/commits/1.17-hunter

Then open pull request with these changes.

**Comment by t0p4** [comment]
Update tomcrypt to 1.18.2 #4

File: cmake/projects/tomcrypt/hunter.cmake (hunk 1 above)
Comment (ruslo) [inline cmake/projects/tomcrypt/hunter.cmake]: Please use https://github.com/hunter-packages/libtomcrypt/releases/tag/v1.18.2-p0 here.

[the remark this answers is not in the document]
Comment (t0p4) [reply cmake/projects/tomcrypt/hunter.cmake]: have done it

File: cmake/projects/tomcrypt/hunter.cmake [hunk 2]
```diff
@@ -41,6 +41,16 @@ hunter_add_version(
...
 )
+hunter_add_version(
+    PACKAGE_NAME
+    tomcrypt
+    VERSION
+    "1.18.2"
+    URL
+    "https://github.com/hunter-packages/libtomcrypt/archive/v1.18.2-p0.tar.gz"
+    SHA1
+    8ACF74D6A3AF6C4360E665A5DF0BA84CA3568D0E
```
Comment (ruslo) [inline cmake/projects/tomcrypt/hunter.cmake]: Lower-case used here in other similar places, let's keep things consistent.

File: cmake/projects/tomcrypt/hunter.cmake [hunk 3]
```diff
@@ -41,6 +41,16 @@ hunter_add_version(
     SHA1
     69b882eff5bf69a5a4e0e9169cc9b9c0ea5b9d71
 )
+hunter_add_version(
+    PACKAGE_NAME
+    tomcrypt
+    VERSION
+    "1.18.2"
```
Comment (ruslo) [inline cmake/projects/tomcrypt/hunter.cmake]: Let's make it the same as tag, please use `1.18.2-p0` here.

File: cmake/configs/default.cmake [hunk 4]
```diff
@@ -387,7 +387,7 @@ hunter_default_version(thrift VERSION 0.11.0-p0)
 hunter_default_version(tinydir VERSION 1.2-p0)
 hunter_default_version(tinyxml2 VERSION 6.2.0-p1)
 hunter_default_version(toluapp VERSION 1.0.93-p1)
-hunter_default_version(tomcrypt VERSION 1.17-p3)
+hunter_default_version(tomcrypt VERSION 1.18.2)
```
Comment (ruslo) [inline cmake/configs/default.cmake]: Please use `1.18.2-p0`

File: cmake/projects/tomcrypt/hunter.cmake [hunk 5]
```diff
@@ -41,6 +41,16 @@ hunter_add_version(
...
 )
+hunter_add_version(
+    PACKAGE_NAME
+    tomcrypt
+    VERSION
+    "1.18.2-p0"
+    URL
+    "https://github.com/hunter-packages/libtomcrypt/archive/v1.18.2-p0.tar.gz"
+    SHA1
+    8acf74d6a3af6c4360e665a5df0ba84ca3568d0e
```
Comment (ruslo) [inline cmake/projects/tomcrypt/hunter.cmake]: SHA1 value is not correct. Note that you will discover this fact if you will run tests properly.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/builds/19372508
* https://travis-ci.org/ingenue/hunter/builds/439156918?utm_source=github_status&utm_medium=notification

File: cmake/configs/default.cmake [hunk 6]
```diff
@@ -387,7 +387,7 @@ hunter_default_version(thrift VERSION 0.11.0-p0)
 hunter_default_version(tinydir VERSION 1.2-p0)
 hunter_default_version(tinyxml2 VERSION 6.2.0-p1)
 hunter_default_version(toluapp VERSION 1.0.93-p1)
-hunter_default_version(tomcrypt VERSION 1.17-p3)
+hunter_default_version(tomcrypt VERSION 1.18.2-p0)
```
Comment (ruslo) [inline cmake/configs/default.cmake]: We got conflict here after merge of https://github.com/ruslo/hunter/pull/1574, please resolve.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/builds/19374300
* https://travis-ci.org/ingenue/hunter/builds/439184387?utm_source=github_status&utm_medium=notification

File: cmake/configs/default.cmake [hunk 7]
```diff
@@ -387,8 +387,8 @@ hunter_default_version(thrift VERSION 0.11.0-p0)
 hunter_default_version(tinydir VERSION 1.2-p0)
 hunter_default_version(tinyxml2 VERSION 6.2.0-p1)
 hunter_default_version(toluapp VERSION 1.0.93-p1)
-hunter_default_version(tomcrypt VERSION 1.17-p3)
-hunter_default_version(tommath VERSION 1.0.1-p0)
+hunter_default_version(tomcrypt VERSION 1.18.2-p0)
+hunter_default_version(tommath VERSION 1.0-p2)
```
Comment (ruslo) [inline cmake/configs/default.cmake]: Why?

Comment (t0p4) [reply cmake/configs/default.cmake]: Why are  macOS builds broken?

Comment (ruslo) [reply cmake/configs/default.cmake]: > hunter_default_version(tommath VERSION 1.0-p2)

Please don't touch `tommath`, this pull request is about `tomcrypt`.

Comment (t0p4) [reply cmake/configs/default.cmake]: ok

Comment (t0p4) [reply cmake/configs/default.cmake]: (tommath VERSION 1.0-p2) ok

Comment (ruslo) [reply cmake/configs/default.cmake]: > (tommath VERSION 1.0-p2) ok

Can you please clarify what you mean?

Comment (t0p4) [reply cmake/configs/default.cmake]: i've  changed it by accident at merge

Comment (t0p4) [reply cmake/configs/default.cmake]: We see eye to eye

**Comment by ruslo** [comment]
This pull request is a big mess now, I will close it.

Please open new pull request, follow [documentation](https://docs.hunter.sh/en/latest/creating-new/update.html) carefully and fill accurately [pull request template](https://raw.githubusercontent.com/ruslo/hunter/master/.github/PULL_REQUEST_TEMPLATE.md). Thanks.

**Pull request closed without merging**